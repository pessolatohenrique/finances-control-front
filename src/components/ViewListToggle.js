import { Button, Grid } from "@mui/material";
import ListIcon from "@material-ui/icons/List";
import ViewModuleIcon from "@material-ui/icons/ViewModule";

function ViewListToggle({ isTable, isList, switchFormat }) {
  return (
    <Grid container justifyContent="flex-end" gap="10px">
      <Button
        data-testid="view-table"
        variant={isTable() ? "contained" : "text"}
        onClick={() => switchFormat("table")}
      >
        <ListIcon />
      </Button>
      <Button
        data-testid="view-card"
        variant={isList() ? "contained" : "text"}
        onClick={() => switchFormat("list")}
      >
        <ViewModuleIcon />
      </Button>
    </Grid>
  );
}

export default ViewListToggle;
