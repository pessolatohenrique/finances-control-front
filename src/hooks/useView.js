import { useState } from "react";

function useView() {
  const [format, setFormat] = useState("table");

  function switchFormat(format) {
    setFormat(format);
  }

  function isTable() {
    return format === "table";
  }

  function isList() {
    return format === "list";
  }

  return [isTable, isList, switchFormat];
}

export default useView;
