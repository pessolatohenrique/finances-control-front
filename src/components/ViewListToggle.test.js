import { render, screen, fireEvent, waitFor } from "@testing-library/react";
import { renderHook } from "@testing-library/react-hooks";
import ViewListToggle from "./ViewListToggle";
import useView from "../hooks/useView";

describe("ViewListToggle", () => {
  it("should generate snapshot", () => {
    const { result } = renderHook(() => useView());
    const [isTable, isList, switchFormat] = result.current;

    const { container } = render(
      <ViewListToggle
        isTable={isTable}
        isList={isList}
        switchFormat={switchFormat}
      />
    );

    expect(container.firstChild).toMatchSnapshot();
  });

  it("should change style on table click", () => {
    const { result } = renderHook(() => useView());
    const [isTable, isList, switchFormat] = result.current;

    render(
      <ViewListToggle
        isTable={isTable}
        isList={isList}
        switchFormat={switchFormat}
      />
    );

    const tableBtn = screen.getByTestId("view-table");
    fireEvent.click(tableBtn);

    expect(screen.getByTestId("view-table")).toHaveClass(
      "MuiButton-containedPrimary"
    );
  });

  it("should change style on card click", () => {
    const { result } = renderHook(() => useView());
    const [isTable, isList, switchFormat] = result.current;

    render(
      <ViewListToggle
        isTable={isTable}
        isList={isList}
        switchFormat={switchFormat}
      />
    );

    const cardBtn = screen.getByTestId("view-card");

    fireEvent.click(cardBtn);

    waitFor(() => {
      expect(screen.getByTestId("view-card")).toHaveClass(
        "MuiButton-containedPrimary"
      );
    });
  });
});
