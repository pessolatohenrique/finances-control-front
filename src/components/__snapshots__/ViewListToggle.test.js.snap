// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ViewListToggle should generate snapshot 1`] = `
<div
  class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-undefined css-1mx4t5n-MuiGrid-root"
>
  <button
    class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root css-sghohy-MuiButtonBase-root-MuiButton-root"
    data-testid="view-table"
    tabindex="0"
    type="button"
  >
    <svg
      aria-hidden="true"
      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
      data-testid="ListIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M3 13h2v-2H3v2zm0 4h2v-2H3v2zm0-8h2V7H3v2zm4 4h14v-2H7v2zm0 4h14v-2H7v2zM7 7v2h14V7H7z"
      />
    </svg>
    <span
      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
    />
  </button>
  <button
    class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root css-1e6y48t-MuiButtonBase-root-MuiButton-root"
    data-testid="view-card"
    tabindex="0"
    type="button"
  >
    <svg
      aria-hidden="true"
      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
      data-testid="ViewModuleIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M4 11h5V5H4v6zm0 7h5v-6H4v6zm6 0h5v-6h-5v6zm6 0h5v-6h-5v6zm-6-7h5V5h-5v6zm6-6v6h5V5h-5z"
      />
    </svg>
    <span
      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
    />
  </button>
</div>
`;
